src/GraphQL/Language/AST/Variable.cs

```diff
@@ -1,20 +1,41 @@
+#nullable enable
+
+using System;
+
 namespace GraphQL.Language.AST
 {
     /// <summary>
     /// Represents a variable name and value tuple that has been gathered from the document and attached <see cref="Inputs"/>.
     /// </summary>
     public class Variable
     {
+        [Obsolete]
+        public Variable() : this(null!)
+        {
+        }
+
+        public Variable(string name)
```

Comment:
It is possible that the best solution would be to use init-only properties.

Reply:
I think I had come to the conclusion that it would be a breaking change.  But for v5, sure.